library/spdm_requester_lib/libspdm_req_set_certificate.c

```diff
@@ -87,11 +94,33 @@ static libspdm_return_t libspdm_try_set_certificate(libspdm_context_t *spdm_cont
     spdm_request->header.param1 = slot_id;
     spdm_request->header.param2 = 0;
 
+    if (spdm_request->header.spdm_version >= SPDM_MESSAGE_VERSION_13) {
+        if ((request_attribute & SPDM_GET_CERTIFICATE_REQUEST_SLOT_ID_MASK) != slot_id) {
```

Comment:
((request_attribute & SPDM_GET_CERTIFICATE_REQUEST_SLOT_ID_MASK) != 0)

Reply:
Thanks. I have changed the code.